cocotb/regression.py

```diff
@@ -124,8 +124,10 @@ def initialise(self):
         self._dut = cocotb.handle.SimHandle(handle) if handle else None
 
         if self._dut is None:
-            raise AttributeError("Can not find Root Handle (%s)" %
```

Comment:
I guess this is going to raise a few eyebrows without an explanation.

Reply:
Ahh... Perhaps I should remove this from the task-based BFM PR. I added this to allow cocotb to run with mainline Verilator (4.022). Mainline Verilator doesn't support full VPI, but still works quite nicely with task-based BFMs as long as cocotb doesn't error out due to not being able to access signals.

Reply:
Added an explanatory comment, and updated the message to better-describe the impact of not having access to the design root.

Reply:
This should at least log.

I'm thinking we should only allow this to continue after checking the simulator environment variable for whether this is verilator.

Reply:
Hmm... that's a good idea. Let me have a look to see how feasible it would be to get sim-info at this point

Reply:
I've changed this to check the SIM_NAME (VPI 'product') and report the current note if the simulator is Verilator, and retain the previous exception if not.